### PR TITLE
feat: add min/max card purchase limits

### PR DESCRIPTION
**Pull request by HashEngineering**
Other changes:
1. credentials file is now called `credentials.json`
2. quiet mode added to block Slack notifications for dev/testing

### CODE REVIEW

**Review by Syn-McJ — APPROVED** [review]
Excellent

File: src/main/kotlin/org/dash/mobile/explore/sync/GCManager.kt [hunk 1]
```diff
@@ -19,14 +19,18 @@ private const val GCS_LOCK_FILE_NAME = "explore/.lock"
...
     private val logger = LoggerFactory.getLogger(GCManager::class.java)!!
 
     private val gcStorage by lazy {
         createStorage()
     }
 
+    private val lockFileName by lazy {
+        "$GCS_LOCK_FILE_NAME-$mode"
+    }
+
```
Comment (HashEngineering) [inline src/main/kotlin/org/dash/mobile/explore/sync/GCManager.kt:33]: each mode will have it's own lock file, which will prevent the production build from running if the testnet or dev build is running.

File: src/main/kotlin/org/dash/mobile/explore/sync/process/DashDirectDataSource.kt [hunk 2]
```diff
@@ -228,6 +228,16 @@ class DashDirectDataSource(private val devApi: Boolean, slackMessenger: SlackMes
...
+
+            minCardPurchase = convertJsonData("MinimumCardPurchase", merchantData)
+            maxCardPurchase = convertJsonData("MaximumCardPurchase", merchantData)
+            savingsPercentage = convertJsonData("SavingsPercentage", merchantData)
+            if (savingsPercentage == null || savingsPercentage == 0.0) {
+                val level1: Double? = convertJsonData("Level1", merchantData)
+                if (level1 != null) {
+                    savingsPercentage = level1
+                }
+            }
```
Comment (HashEngineering) [inline src/main/kotlin/org/dash/mobile/explore/sync/process/DashDirectDataSource.kt:240]: The DashDirect people said that SavingsPercentage would replace the "Levelx" fields, but it hasn't happened yet.  

This code block will allow for both old and new API formats to get the discount.

**Review by Syn-McJ — APPROVED** [review]
Great improvements

**Pull request merged**